connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -231,6 +265,53 @@ public void execute() {
         }
     }
 
+    protected List<SourceRecord> poll() throws InterruptedException {
+        try {
+            return task.poll();
+        } catch (RetriableException e) {
+            log.error("Encountered retriable error on polling task", e);
+            // Do nothing. Let the framework poll whenever it's ready.
+            return null;
+        }
+    }
+
+    protected <V> V execute(Operation<V> operation) {
+        return operationExecutor.execute(operation, processingContext).result();
+    }
+
+    /**
+     * Convert the source record into a producer record.
+     *
+     * @param record the transformed record
+     * @return null, if the input is null, or an error was encountered during any of the converter stages.
+     */
+    private ProducerRecord<byte[], byte[]> convertTransformedRecord(SourceRecord record) {
+        if (record == null) {
+            return null;
+        }
+
+        processingContext.setCurrentContext(Stage.HEADER_CONVERTER, headerConverter.getClass());
+        RecordHeaders headers = execute(() -> convertHeaderFor(record));
+        if (processingContext.failed()) {
```

Comment:
I think these conditions are no longer required. Probably needs condition in the last step alone?